File: lib/core/models/progress_info.dart
/// معادل ProgressInfo.cs (نسخه‌ی ساده‌شده برای شروع)
class ProgressInfo {
  final String currentFile;
  final double percentage; // 0 تا 100
  final String status;

  ProgressInfo({
    this.currentFile = '',
    this.percentage = 0,
    this.status = '',
  });
}

File: windows/SFE.Core/Models/ProgressInfo.cs
namespace SFE.Core.Models;

public class ProgressInfo
{
    public string CurrentFile { get; set; } = "";
    public int TotalFiles { get; set; }
    public int ProcessedFiles { get; set; }
    public double Percentage { get; set; }
    public string Status { get; set; } = "";
    public double BytesPerSecond { get; set; }
    public TimeSpan? TimeRemaining { get; set; }
    public long TotalBytes { get; set; }
    public long ProcessedBytes { get; set; }
    public string? CompletedPath { get; set; }
    public bool? CompletedSuccess { get; set; }
}